### DISCUSSION TITLE
CVE-2023-25330 Scanning misses

### DISCUSSION THREAD

**Discussion by zangcc** [aquasecurity/trivy#5623] [Q&A] [answered]
### Description

CVE-2023-25330 Scanning misses。According to the official description:A SQL injection vulnerability in Mybatis plus below 3.5.3.1 allows remote attackers to execute arbitrary SQL commands via the tenant ID valuer..

![image](https://github.com/aquasecurity/trivy/assets/64825932/cb524fd1-e75f-4025-b7e6-01bad6e9c657)

My version is 3.5.1, which is a vulnerable version, but trivy did not scan it.

`<dependency>
         <groupId>com.baomidou</groupId>
         <artifactId>mybatis-plus-extension</artifactId>
        <version>3.5.1</version>
</dependency>`

![image](https://github.com/aquasecurity/trivy/assets/64825932/3073091f-35ea-4f24-8ce9-342682565c51)

### Desired Behavior

trivy can correctly scan the vulnerable version

### Actual Behavior

Unable to scan Mybatis plus below 3.5.3.1

### Reproduction Steps

```bash
1./root/tools/trivy/trivy_0.47.0/trivy  fs /root/code/code_project/pom.xml  --skip-db-update --offline-scan  --dependency-tree --timeout 999m
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
2023-11-21T19:42:00.388+0800	INFO	"--dependency-tree" only shows the dependents of vulnerable packages. Note that it is the reverse of the usual dependency tree, which shows the packages that depend on the vulnerable package. It supports limited package managers. Please see the document for the detail.
2023-11-21T19:42:00.388+0800	DEBUG	'--dependency-tree' enables '--list-all-pkgs'.
2023-11-21T19:42:00.388+0800	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-11-21T19:42:00.388+0800	DEBUG	Ignore statuses	{"statuses": null}
2023-11-21T19:42:00.390+0800	DEBUG	cache dir:  /root/.cache/trivy
2023-11-21T19:42:00.390+0800	DEBUG	Skipping DB update...
2023-11-21T19:42:00.390+0800	DEBUG	DB Schema: 2, UpdatedAt: 2023-11-21 06:12:44.049179772 +0000 UTC, NextUpdate: 2023-11-21 12:12:44.049179481 +0000 UTC, DownloadedAt: 0001-01-01 00:00:00 +0000 UTC
2023-11-21T19:42:00.390+0800	INFO	Vulnerability scanning is enabled
2023-11-21T19:42:00.390+0800	DEBUG	Vulnerability type:  [os library]
2023-11-21T19:42:00.390+0800	INFO	Secret scanning is enabled
2023-11-21T19:42:00.390+0800	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-11-21T19:42:00.390+0800	INFO	Please see also https://aquasecurity.github.io/trivy/v0.47/docs/scanner/secret/#recommendation for faster secret detection
2023-11-21T19:42:00.390+0800	DEBUG	No secret config detected: trivy-secret.yaml
2023-11-21T19:42:00.390+0800	DEBUG	The nuget packages directory couldn't be found. License search disabled
2023-11-21T19:42:00.390+0800	DEBUG	Walk the file tree rooted at '/root/code/demo-gm-transaction/pom1.xml' in parallel
2023-11-21T19:42:00.395+0800	DEBUG	OS is not detected.
2023-11-21T19:42:00.395+0800	DEBUG	Detected OS: unknown
2023-11-21T19:42:00.395+0800	INFO	Number of language-specific files: 0
```

### Operating System

Ubuntu SMP Fri Mar 18 12:42:08 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Version

```bash
Version: 0.47.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-11-21 06:12:44.049179772 +0000 UTC
  NextUpdate: 2023-11-21 12:12:44.049179481 +0000 UTC
  DownloadedAt: 0001-01-01 00:00:00 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-07-24 00:52:18.487469647 +0000 UTC
  NextUpdate: 2023-07-27 00:52:18.487469147 +0000 UTC
  DownloadedAt: 2023-07-24 07:52:17.429868261 +0000 UTC
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by nikpivkin** [comment]
As I understand from the [vulnerability description](https://github.com/FCncdn/MybatisPlusTenantPluginSQLInjection-POC/blob/master/Readme.en.md), the sql injection affects the `com.baomidou:mybatis-plus-extension` package, not `com.baomidou:mybatis-plus`, since the sql parser is in the `mybatis-plus-extension` artifact. Therefore, the affected package is not quite correct. @DmitriyLewen Wdyt?

**Reply by zangcc** [reply]
Thank you for your reply, so why is my "com.baomidou:mybatis-plus" dependency version 3.5.3.1 recognized by Trivy? However, the vulnerable version 3.5.1 cannot be scanned. (cve-2023-25330)

<img width="933" alt="image" src="https://github.com/aquasecurity/trivy/assets/64825932/fd6eea74-82be-4da8-ae62-735e510e1c9f">

@nikpivkin

**Reply by DmitriyLewen** [reply]
Hello @zangcc 

I am confused what you expected in this case.
Only `mybatis-plus-boot-starter` uses `mybatis-plus` dependency:
```bash
➜ mvn dependency:tree | grep mybatis-plus 
[INFO] +- com.baomidou:mybatis-plus-extension:jar:3.5.1:compile
[INFO] |  +- com.baomidou:mybatis-plus-core:jar:3.5.1:compile
[INFO] |  |  +- com.baomidou:mybatis-plus-annotation:jar:3.5.1:compile
[INFO] +- com.baomidou:mybatis-plus-boot-starter:jar:3.5.1:compile
[INFO] |  +- com.baomidou:mybatis-plus:jar:3.5.1:compile
[INFO] \- com.baomidou:mybatis-plus-generator:jar:3.5.1:compile
```

Trivy finds this CVE correctly:
```bash
➜ trivy fs ./pom.xml | grep CVE-2023-25330
...
│ com.baomidou:mybatis-plus                          │ CVE-2023-25330 │ CRITICAL │ affected │ 3.5.1             │                               │ MyBatis-Plus vulnerable to SQL injection via TenantPlugin    │
```

If you're concerned that Trivy isn't showing the corrected version - I'm not sure the description is correct.
Take a look on the only reference - https://github.com/FCncdn/MybatisPlusTenantPluginSQLInjection-POC/blob/master/Readme.en.md:
```markdown
MavenGroupId: com.baomidou

MavenArtifactId: mybatis-plus

AffectedVersion: 3.x
```
But if you have information about a fix in version 3.5.3.1 (PR, commit or whatever) - create changes in GitHub please - https://github.com/advisories/GHSA-32qq-m9fh-f74w/improve

Regards, Dmitriy

**Reply by zangcc** [reply]
This is really weird. Why do dependencies in my pom.xml file:
"mybatis-plus-boot-starter 3.5.1" cannot be scanned by trivy?

![image](https://github.com/aquasecurity/trivy/assets/64825932/c17c4f87-d276-406f-8f94-283aa1b1b358)

![image](https://github.com/aquasecurity/trivy/assets/64825932/889fed38-840e-4382-9976-52c8083fc3d1)

@DmitriyLewen

**Reply by DmitriyLewen** [reply]
you use the `--offline-scan` flag.
Do you have `mybatis-plus-boot-starter` in your .m2 folder?

**Reply by zangcc** [reply]
mybatis-plus-boot-starter 3.5 1 dependency does exist.  Still can't scan it

![image](https://github.com/aquasecurity/trivy/assets/64825932/cdf11222-1fdf-4176-a47a-53525f44e77b)

**Reply by DmitriyLewen** [reply]
Okay. Looks like Trivy can't find your local repository.
Can you say how you change local repository?

**Reply by zangcc** [reply]
`/root/tools/apache-maven-3.9.2//bin/mvn    -gs /root/tools/apache-maven-3.9.2/settings.xml  clean install   -Dmaven.test.skip -Dmaven.repo.local=/root/tools/apache-maven-3.9.2/repository`

**Reply by zangcc** [reply]
Thank you very much, you are indeed right. I changed the local repositor to .m2 and scanned it again.

The problem has been solved
@DmitriyLewen

**Reply by DmitriyLewen** [reply, marked as answer]
Some more information for you and users which will watch this discussion:
by default there are 2 places for `settings.xml` file (https://maven.apache.org/settings.html#quick-overview):
- ${maven.home}/conf/settings.xml
- ${user.home}/.m2/settings.xml

If you use `-gs`, Trivy will not be able to find this file.
Better to use default folders for your settings.

**Comment by ba1ma0** [comment]
Hi @DmitriyLewen D  It seems that there is NO Vulnerability CVE-2023-25330  in  com.baomidou:mybatis-plus 3.5.4.1 according to Maven official but trivy found a  CVE-2023-25330  in com.baomidou:mybatis-plus 3.5.4.1 .  I am wondering is there any mistake in trivy
https://mvnrepository.com/artifact/com.baomidou/mybatis-plus
![image](https://github.com/aquasecurity/trivy/assets/32502627/17fff0da-b997-465e-a143-b18258f62055)
![image](https://github.com/aquasecurity/trivy/assets/32502627/4e6f3120-4512-4836-a92d-933876c61080)
